### PR TITLE
refactor TableExtractor to merge two extractTablesFromInsert methods

### PR DESCRIPTION
**Pull request by cheese8**
Ref #17431. This PR based on #17432 , please review #17432 first, thanks.

Changes proposed in this pull request:
-refactor TableExtractor to merge two extractTablesFromInsert methods

### CODE REVIEW

File: shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/statement/dml/InsertStatementContextUtil.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+     */
+    public static List<InsertStatement> getInsertStatements(final InsertStatement sqlStatement) {
+        Optional<InsertMultiTableElementSegment> optional = getInsertMultiTableElementSegment(sqlStatement);
+        if (optional.isPresent()) {
+            return new LinkedList<>(optional.get().getInsertStatements());
+        }
+        return Collections.singletonList(sqlStatement);
+    }
+    
+    private static Optional<InsertMultiTableElementSegment> getInsertMultiTableElementSegment(final InsertStatement sqlStatement) {
```
Comment (strongduanmu) [inline shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/statement/dml/InsertStatementContextUtil.java:59]: @cheese8 Please move this logic to InsertStatementHandler.

File: shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/statement/dml/InsertStatementContextUtil.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+ */
+public final class InsertStatementContextUtil {
+    
+    /**
+     * Get first index on MultiInsertStatement.
+     * @param sqlStatementContext InsertStatementContext
+     * 
+     * @return column names collection
+     */
+    public static SimpleTableSegment getTable(final InsertStatementContext sqlStatementContext) {
```
Comment (strongduanmu) [inline shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/statement/dml/InsertStatementContextUtil.java:41]: @cheese8 Can we remove InsertStatementContextUtil class, and move  getTable method to InsertStatementContext.

File: shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/statement/dml/InsertStatementContext.java [hunk 3]
```diff
@@ -62,38 +63,51 @@ public final class InsertStatementContext extends CommonSQLStatementContext<Inse
...
-        insertSelectContext = getInsertSelectContext(metaDataMap, parameters, parametersOffset, defaultDatabaseName).orElse(null);
-        onDuplicateKeyUpdateValueContext = getOnDuplicateKeyUpdateValueContext(parameters, parametersOffset).orElse(null);
+        
         tablesContext = new TablesContext(getAllSimpleTableSegments(), getDatabaseType());
         ShardingSphereSchema schema = getSchema(metaDataMap, defaultDatabaseName);
-        columnNames = useDefaultColumns() ? schema.getAllColumnNames(sqlStatement.getTable().getTableName().getIdentifier().getValue()) : insertColumnNames;
-        generatedKeyContext = new GeneratedKeyContextEngine(sqlStatement, schema)
-                .createGenerateKeyContext(insertColumnNames, getAllValueExpressions(sqlStatement), parameters).orElse(null);
+        
+        AtomicInteger parametersOffset = new AtomicInteger(0);
```
Comment (strongduanmu) [inline shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/statement/dml/InsertStatementContext.java:96]: Please add unit test for this new logic.

File: shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/statement/dml/InsertStatementContext.java [hunk 4]
```diff
@@ -62,38 +63,51 @@ public final class InsertStatementContext extends CommonSQLStatementContext<Inse
...
+        List<InsertStatement> insertStatements = InsertStatementContextUtil.getInsertStatements(sqlStatement);
+        for (int cursor = 0; cursor < insertStatements.size(); cursor++) {
```
Comment (strongduanmu) [inline shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/statement/dml/InsertStatementContext.java:98]: index may be better.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/statement/dml/InsertStatementContextUtil.java (hunk 1 above)
Comment (cheese8) [reply shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/statement/dml/InsertStatementContextUtil.java:59]: This PR based on #17432 , already updated in it.

File: shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/statement/dml/InsertStatementContextUtil.java (hunk 2 above)
Comment (cheese8) [reply shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/statement/dml/InsertStatementContextUtil.java:41]: This PR based on #17432 , already updated in it.

File: shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/statement/dml/InsertStatementContext.java (hunk 3 above)
Comment (cheese8) [reply shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/statement/dml/InsertStatementContext.java:96]: This PR based on #17432 , already updated in it.

**Pull request closed without merging**